File: Components/app/src/main/dotnet/app/domain/weatherforecast/handlers/GetWeatherForecastHandler.cs
using System;
using System.Linq;
using System.Text;
using System.Text.Json;
using System.Threading;
using System.Threading.Tasks;
using MediatR;
using Microsoft.Extensions.Caching.Distributed;
using Microsoft.Extensions.Logging;
using se.clouds.javanet.app.domain.weatherforecast.queries;

namespace se.clouds.javanet.app.domain.weatherforecast.handlers
{
    public class GetWeatherForecastHandler : IRequestHandler<GetWeatherForecast, WeatherForecast[]>
    {
        private static readonly string[] Summaries = new[]
        {
            "Freezing", "Bracing", "Chilly", "Cool", "Mild", "Warm", "Balmy", "Hot", "Sweltering", "Scorching"
        };
        private readonly ILogger<GetWeatherForecastHandler> _logger;
        private readonly IDistributedCache _cache;
        public TimeSpan CachedTime { get; set; } = TimeSpan.FromHours(12);
        public GetWeatherForecastHandler(ILogger<GetWeatherForecastHandler> logger, IDistributedCache cache)
        {
            _logger = logger;
            _cache =  cache;
        }
        public async Task<WeatherForecast[]> Handle(GetWeatherForecast request, CancellationToken cancellationToken)
        {
            WeatherForecast[] result;
            var keyId ="WeatherForecast-0";

            var cacheResp = await _cache.GetAsync(keyId, cancellationToken);
            if (cacheResp != null)
            {
                _logger.LogDebug($"--- Redis: {Encoding.UTF8.GetString(cacheResp)}");
                return JsonSerializer.Deserialize<WeatherForecast[]>(Encoding.UTF8.GetString(cacheResp));
            }
            else {
                _logger.LogDebug($"--- No cache entry could be found");
                var rng = new Random();

                result = Enumerable.Range(1, 5).Select(index => new WeatherForecast
                {
                    Date = DateTime.Now.AddDays(index),
                    TemperatureC = rng.Next(-20, 55),
                    Summary = Summaries[rng.Next(Summaries.Length)]
                })
                .ToArray();

                var serialized = JsonSerializer.Serialize<WeatherForecast[]>(result);
                var options = new DistributedCacheEntryOptions().SetSlidingExpiration(CachedTime);
                await _cache.SetAsync(keyId, Encoding.UTF8.GetBytes(serialized), options, cancellationToken);
            }

            return result;
        }

    }
}

File: Components/app/src/main/java/se/clouds/javanet/app/domain/weatherforecast/handler/GetWeatherForecastHandler.java
package se.clouds.javanet.app.domain.weatherforecast.handler;

import java.util.Random;
import java.util.stream.Collectors;
import org.json.JSONArray;
import org.json.JSONObject;
import se.clouds.javanet.app.domain.weatherforecast.WeatherForecast;
import se.clouds.javanet.app.domain.weatherforecast.query.GetWeatherForecast;
import se.clouds.javanet.core.controller.ActionResult;
import se.clouds.javanet.core.controller.IActionResult;
import se.clouds.javanet.core.di.Di;
import se.clouds.javanet.core.mediator.IRequestHandler;
import se.clouds.javanet.core.mediator.MediatR;
import se.clouds.javanet.core.mediator.Task;
import se.clouds.javanet.core.controller.ResultStatus;


public class GetWeatherForecastHandler implements IRequestHandler<GetWeatherForecast, IActionResult> {

    //private MediatR<IActionResult> mediatr = (MediatR)Di.GetSingleton(IMediator.class, MediatR.class);
    private static String[] Summaries = new String[]
    {
        "Freezing", "Bracing", "Chilly", "Cool", "Mild", "Warm", "Balmy", "Hot", "Sweltering", "Scorching"
    };
    public GetWeatherForecastHandler() {
        var mediatr = ((MediatR<IActionResult>)Di.GetMediator());
        mediatr.addRequestHandler(new GetWeatherForecast(), new HandlerTask());
    }

    public class HandlerTask implements Task<IActionResult>
    {
        @Override
        public IActionResult call() throws Exception
        {
            var result = new ActionResult();
            JSONArray arr = new JSONArray();

            new Random().ints(5, -20, 50)
                .mapToObj(rnd ->
                    arr.put(new JSONObject(
                        new WeatherForecast(rnd, Summaries[new Random().nextInt((Summaries.length))])
                            .AsMap()))).collect(Collectors.toList());

            result.SetContent(JSONObject.valueToString(arr));
            result.SetStatus(ResultStatus.Status.OK);
            return result;
        }
    }

}
